src/app/models/AudioRecording.ts

```diff
@@ -86,9 +90,33 @@ export class AudioRecording
   public site?: Site;
 
   public get viewUrl(): string {
+    return this.playUrl;
+  }
+
+  public get detailsUrl(): string {
+    return this.getDetailsUrl();
+  }
+
+  public getDetailsUrl(
+    project?: IdOr<Project>,
+    region?: IdOr<Region>,
+    site?: IdOr<Site>
+  ): string {
+    return audioRecordingMenuItem.route.format(
+      { audioRecordingId: this.id },
+      { projectId: id(project), regionId: id(region), siteId: id(site) }
+    );
+  }
+
+  public get playUrl(): string {
     return listenRecordingMenuItem.route.format({ audioRecordingId: this.id });
   }
 
+  public get downloadUrl(): string {
+    // TODO Add download url when download page built
+    throw new Error("not implemented");
+  }
+
```

Comment:
you mean this to be the view url to the download batch page right?
It should be named batchDownloadUrl then